mirgecom/multiphysics/thermally_coupled_fluid_wall.py

```diff
@@ -150,13 +174,78 @@ def get_external_grad_av(self, discr, dd_bdry, grad_av_minus, **kwargs):
         # Grab a unit normal to the boundary
         nhat = thaw(discr.normal(dd_bdry), actx)
 
+        # REVIEW QUESTION
         # Apply a Neumann condition on the energy gradient
-        ext_grad_energy = (
-            grad_av_minus.energy - 2 * np.dot(grad_av_minus.energy, nhat) * nhat)
+        # This also looks suspiscous: I blv the "2" should not be there
+        # for this term.
+        ext_grad_energy = \
+            grad_av_minus.energy - 2 * np.dot(grad_av_minus.energy, nhat) * nhat
+
+        # REVIEW QUESTION
+        # uh oh - we don't have the necessary data to compute grad_y from grad_av
+        # from mirgecom.fluid import species_mass_fraction_gradient
+        # grad_y_minus = species_mass_fraction_gradient(state_minus.cv,
+        #                                               grad_cv_minus)
+        # grad_y_plus = grad_y_minus - np.outer(grad_y_minus@normal, normal)
+        # grad_species_mass_plus = 0.*grad_y_plus
+        # This re-stuffs grad_y+ back into grad_cv+, skipit; we did not split AVs
+        # for i in range(state_minus.nspecies):
+        #    grad_species_mass_plus[i] = (state_minus.mass_density*grad_y_plus[i]
+        #        + state_minus.species_mass_fractions[i]*grad_cv_minus.mass)
+        ext_grad_species_mass = \
+            grad_av_minus.species_mass - np.outer(grad_av_minus@nhat, nhat)
 
         return make_conserved(
             discr.dim, mass=grad_av_minus.mass, energy=ext_grad_energy,
-            momentum=grad_av_minus.momentum, species_mass=grad_av_minus.species_mass)
+            momentum=grad_av_minus.momentum, species_mass=ext_grad_species_mass)
+
+    def get_external_grad_cv(self, state_minus, grad_cv_minus, normal, **kwargs):
+        """Return grad(CV) to be used in the boundary calculation of viscous flux."""
+        from mirgecom.fluid import species_mass_fraction_gradient
+        grad_y_minus = species_mass_fraction_gradient(state_minus.cv, grad_cv_minus)
+        grad_y_plus = grad_y_minus - np.outer(grad_y_minus@normal, normal)
```

Comment:
This is the same consideration as above.  Since we aren't using some fluxing function F(f_+, f_-) that is a function of both interior and exterior physical fluxes, we directly enforce the correct value (i.e. the BC) of the gradient.   We have an actual test for this with analytical solution.  I'll get it added  - but wasn't planning to add it to this PR.

Reply:
Got it. Maybe for clarity we should label things as `_bc` instead of `_plus` here?

Reply:
> Got it. Maybe for clarity we should label things as `_bc` instead of `_plus` here?

i like the idea of this.   Mengaldo makes this distinction as well.  I'll take a crack at updating the names to see if it helps .

Reply:
([a87fe63](https://github.com/majosm/mirgecom/pull/3/commits/a87fe63a2a4d88d81cf6e713fde725406175e8b6)) Is hopefully some steps in the right direction.